checks/pre-commit

```diff
@@ -0,0 +1,7 @@
+set -e
+echo "precommit running. checking build located at ${subdir}."
+cd ${subdir}
+../checks/run-coverage.sh
```

Comment:
so this script basically just calls make now, which I think will print out enough information for the user to debug their problem - what would and explicit check on the code accomplish?